### ISSUE TITLE
Create [FETCHER] temp_directory if it doesn't exist?

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/mik#120]
Any reason not to do this? We create the [WRITER] output_directory if it doesn't exist. [FILE_GETTER] also defines a temp_directory, and it probably doesn't need to have the same value as [FETCHER]'s, but I can't think of good use case for it not to be, other than you were getting very large quantities of files and you wanted to write them to some specially configured or temporary file space.

If we can't think of any good reasons not to automatically create the directory defined at [FETCHER] [Ftemp_directory I say we do it. I can take on this task.

**Comment by MarcusBarnes** [comment]
@mjordan Please go ahead.   We can document that the specified folder is created automatically if it does not already exist so that people with unique use cases can modify this as required.

**Comment by mjordan** [comment]
@MarcusBarnes I've got this done in a local feature branch, but notice that fetchers don't use any logging. I'd like to add instantiation of a log object to the fetcher abstract class so we can log its inability to create the temp directory (for example). That OK?

**Comment by MarcusBarnes** [comment]
@mjordan Yes, that would be great.  Thank you.

**Comment by mjordan** [comment]
I've pushed this to the issue-120 branch but logging is not working the way I expect. To test, use a temp_directory setting like

```
[FETCHER] temp_directory = "/tmp/issue-120/cartoons_temp"
```

and make `issue-120` not writable by the user running MIK.

When I run MIK, it dies (which is what I expect) but the error message written to mik.log is not the one I implement [in the fetcher child class](https://github.com/MarcusBarnes/mik/blob/issue-120/src/fetchers/Csv.php#L45). The error in the log is:
```
[2016-01-16 09:29:33] ErrorException.ERROR: ErrorException {"message":54,"code":[],"severity":2,"file":"mkdir(): Permission denied","line":"/home/mark/Documents/hacking/mik/src/fetchers/Fetcher.php"} []
```

This is a suitable error message to log, but I am still no fully understanding how to log specific errors in the main classes.

**Comment by MarcusBarnes** [comment]
@mjordan Thank you for your work so far on this.  I'll plan on taking a look at this later this week.  Once I have it working, I should probably create developer documentation on custom log messages.

**Comment by mjordan** [comment]
Awesome, thank you.

**Comment by MarcusBarnes** [comment]
Closed as of commit https://github.com/MarcusBarnes/mik/commit/bd94688b3bf764406b9862b3bfde234a2f4b9076 with pull-request https://github.com/MarcusBarnes/mik/pull/138

The logging is not functioning in the best possible way.  This can be an area for improvement for the 1.0 milestone.  I've updated https://github.com/MarcusBarnes/mik/issues/42 to reflect this.

**Comment by mjordan** [comment]
Reopening as these changes conflict with test code.

**Comment by mjordan** [comment]
I've added logging configuration to all the fetcher tests. Tests all pass now.

**Comment by mjordan** [comment]
Fixes are in branch issue-120.2.

**Comment by mjordan** [comment]
A fix has been merged into master - can we close this issue?
